### DISCUSSION TITLE
Use "react-native-screens" features at "react-navigation"

### DISCUSSION THREAD

**Discussion by Crysp** [software-mansion/react-native-screens#1731] [Q&A] [answered]
Hi 👋

As far as I understand to use "react-native-screens" with "react-navigation" I should use native stack and just install "react-native-screens" package. But how I can use "react-native-screens" options for screens? Like `sheetAllowedDetents` for modals `formSheet`.

**Comment by kkafar** [comment, marked as answer]
Hi @Crysp,

> As far as I understand to use "react-native-screens" with "react-navigation" I should use native stack and just install "react-native-screens" package.

That's right.

> But how I can use "react-native-screens" options for screens?

`react-navigation`'s native stack uses `react-native-screens` primitives under the hood, so when you set props in JS `react-navigation` all the native ones are passed to `react-native-screens` native components. Example: when you set `presentation: 'modal'` on `react-navigation`'s `Screen` it is passed to native screen component as `stackPresentation: 'modal'`.


> Like sheetAllowedDetents for modals formSheet.

the prop you are interested in `sheetAllowedDetents` has not been released yet. Once it is, the `react-navigation` team will surely provide appropriate API to use it. 

Tag me if there are any follow up questions so that I notice.
